jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/CompoundPool.java

```diff
@@ -0,0 +1,147 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.util;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+/**
+ * This implementation does not support multiplexing.
+ *
+ * @param <P> the type of the pooled objects
+ */
+public class CompoundPool<P> implements Pool<P>
+{
+    private final ConcurrentPool<P> concurrentPool;
+    private final Pool<P> secondBestPool;
+
+    public CompoundPool(ConcurrentPool.StrategyType strategyType, int maxSize, boolean cache)
```

Comment:
I considered such constructor, but how to make sure both underlying pools are compatible, especially w.r.t multiplexing?

Reply:
I still think this is the right thing to do, even if we cannot have `isMultiplexing`.     
IT would be better even if the constructor was private an only a static factory method with this signature was available